### PR TITLE
Space - Diana

### PR DESCRIPTION
**Pull request by dnguye2**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | Handles dealing with the database - additions, deletions, updates, etc.
Describe in your own words what the Controller is doing in Rails | Handles controller actions, getting responses, and logic that are defined in routes
Describe in your own words what the View is doing in Rails | Handles the layout and display of content by using `html.erb` files which translates ruby code into html which displays the content
Describe an edge-case controller test you wrote | N/A because it was an optional :)
What is the purpose of using strong params? (i.e. the params method in the controller) | Allows for only certain parameters to be changed. For instance, you would not want someone to have the following access to these parameters `{user: {is_admin: true, id: 123}}` and would use strong params to only allow the user to put in their name not admin rights. More explanation:[[x]](https://stackoverflow.com/questions/39519186/understand-the-require-of-strong-parameters-rails)
How are Rails migrations related to Rails models? | Creating a model will create a migration
Describe one area of Rails that are still unclear on | Best practice for creating your own routes

### CODE REVIEW

File: app/views/tasks/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<main>
+  <h2> List of Tasks</h2>
+
+
+    <% if @tasks.length > 0 %>
+    <ul>
+      <li><%= link_to "Add a task", new_task_path %></li>
```
Comment (kaidamasaki) [inline app/views/tasks/index.html.erb:7]: It would have been nice if this link showed up when there were no tasks as well.

**Comment by kaidamasaki** [comment]
# Task List




## Major Learning Goals/Code Review





| Criteria | yes/no, and optionally any details/lines of code to reference |
| --- | --- |
| At least 6 commits with meaningful commit messages | ✔️
| Routes follow RESTful conventions | ✔️
| Uses named routes (like `_path`) | ✔️
| Creates Models and migrations | ✔️
| Creates styled views | No styling.
| Handles errors like nonexistant tasks | ✔️
| Uses `form_with` to render forms in Rails | ✔️

## Functional Requirements/Manual Testing

| Functional Requirement | yes/no |
| --- | --- |
| Successfully handles index & show | ✔️
| index & show tests pass | ✔️
| Successfully handles: New, Create | ✔️
| New, Create tests pass | ✔️
| Successfully handles: Edit, Update | ✔️
| Successfully handles: Destroy, Task Complete | ✔️

## Overall Feedback

| Overall Feedback | Criteria | yes/no |
| --- | --- | --- |
| Green (Meets/Exceeds Standards) | 5+ in Code Review && 5+ in Functional Requirements | ✔️
| Yellow (Approaches Standards) | 3+ in Code Review && 4+ in Functional Requirements, or the instructor judges that this project needs special attention |
| Red (Not at Standard) | 0-2 in Code Review or 0-3 in Functional Reqs, or assignment is breaking/doesn’t run with less than 5 minutes of debugging, or the instructor judges that this project needs special attention |





## Code Style Bonus Awards



Was the code particularly impressive in code style for any of these reasons (or more...?)

| Quality | Yes? |
| --- | --- |
| Perfect Indentation | ✅
| Elegant/Clever | ✅
| Descriptive/Readable |
| Concise | ✅
| Logical/Organized | ✅
